### PR TITLE
Feature/wdsus 7 implement wp cli

### PR DESCRIPTION
**Pull request by guzmandrade-dev**
Closes #WDSUS-7

Adds initial support for WP-CLI command `wp abs create_portable_block`.

## What type of PR is this? (put an x to all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Link to test

On a terminal run the following WP-CLI command

```sh
wp abs create_portable_block myblock --title='This is myblock' --desc='This block is used for wds.' --keyword='mykeyword' --icon='table-row-before'
```

Which should create a new folder `blocks/myblock` (the command will fail if the folder already exists) with `block.json` and other files used to register the ACF Block.

the block should be available on the block selector under the category `WDS Blocks`:

Screenshot 2023-06-02 at 4 04 54 PM

## Description

`Call To Action` Example (included in this PR) was created with:

```
wp abs create_portable_block call-to-action --title='Call to Action' --desc='This block is used for wds.' --keyword='call-to-action' --icon='megaphone'
```

Edit page:

Screenshot 2023-05-26 at 4 33 56 PM

Frontend and preview:

Screenshot 2023-06-02 at 4 06 39 PM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [Confluence](https://webdevstudios.atlassian.net/wiki/spaces/wds1/pages/2988474566/Feature+Documentation)
- [ ] 🙅 no documentation needed

## Others

- [ ] 🦮 Is this issue accessible? (Section 508/WCAG 2.0AA)
- [X] 🙌 Does this issue pass all the linting? (PHPCS, ESLint, SassLint)

### CODE REVIEW

File: blocks/call-to-action/editor.css
Comment (khleomix) [inline blocks/call-to-action/editor.css:1]: @guzmandrade-wds Styles won't be added as part of the generator.

File: blocks/call-to-action/editor.js
Comment (khleomix) [inline blocks/call-to-action/editor.js:1]: Here as well.

File: blocks/call-to-action/index.js
Comment (khleomix) [inline blocks/call-to-action/index.js:1]: Same here. :)

File: blocks/call-to-action/script.js
Comment (khleomix) [inline blocks/call-to-action/script.js:1]: Here too. Scripts on the other hand will still be part of the block infrastructure.

File: blocks/call-to-action/style.css
Comment (khleomix) [inline blocks/call-to-action/style.css:1]: Same here, all styles will be handled on the theme level.

File: inc/wpcli/block-starter/block.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "wds/{{name}}",
+	"title": "{{title}}",
+	"description": "{{description}}",
+	"editorStyle": "file:./editor.css",
```
Comment (khleomix) [inline inc/wpcli/block-starter/block.json]: This can be removed.

File: inc/wpcli/block-starter/block.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"editorScript": "./editor.js",
+	"style": "file:./style.css",
```
Comment (khleomix) [inline inc/wpcli/block-starter/block.json]: This too.

File: inc/wpcli/block-starter/editor.css
Comment (khleomix) [inline inc/wpcli/block-starter/editor.css:1]: File to be deleted as well.

File: inc/wpcli/block-starter/editor.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import './editor.css';
```
Comment (khleomix) [inline inc/wpcli/block-starter/editor.js]: Here too.

File: inc/wpcli/block-starter/script.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Block script.
+ *
+ * @package
+ * @since 2.0.0
+ */
+import './style.css';
```
Comment (khleomix) [inline inc/wpcli/block-starter/script.js]: And here

File: inc/wpcli/block-starter/style.css
Comment (khleomix) [inline inc/wpcli/block-starter/style.css:1]: Remove this file too.

File: inc/wpcli/class-blocks-scaffold.php [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+	/**
+	 * Create the block editor styles.
+	 *
+	 * @since 2.0.0
+	 * @author Biplav Subedi <biplav.subedi@webdevstudios.com>
+	 */
+	private function create_block_editor_assets() {
+		$assets_js  = ROOT_PATH . 'inc/wpcli/block-starter/editor.js';
+		$assets_css = ROOT_PATH . 'inc/wpcli/block-starter/editor.css';
```
Comment (khleomix) [inline inc/wpcli/class-blocks-scaffold.php]: Remove the style references from this function. We only need the js.

File: inc/wpcli/class-blocks-scaffold.php [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+	/**
+	 * Create the block main styles.
+	 *
+	 * @since 2.0.0
+	 * @author Biplav Subedi <biplav.subedi@webdevstudios.com>
+	 */
+	private function create_block_assets() {
+		$assets_js  = ROOT_PATH . 'inc/wpcli/block-starter/script.js';
+		$assets_css = ROOT_PATH . 'inc/wpcli/block-starter/style.css';
```
Comment (khleomix) [inline inc/wpcli/class-blocks-scaffold.php]: Remove the style references from this function. We only need the js.

**Review by khleomix — CHANGES_REQUESTED** [review]
@guzmandrade-wds Thanks! I'ved added in some comments, we will need to remove any styles from the generator. Also there's a function that was added by Inna: `register_wds_category` if you can hook into that so that any generated block will automatically be added to that category. Thanks!

File: blocks/call-to-action/script.js
Comment (guzmandrade-dev) [reply blocks/call-to-action/script.js:1]: 👍

File: blocks/call-to-action/editor.css
Comment (guzmandrade-dev) [reply blocks/call-to-action/editor.css:1]: 👍

File: blocks/call-to-action/editor.js
Comment (guzmandrade-dev) [reply blocks/call-to-action/editor.js:1]: 👍

File: blocks/call-to-action/index.js
Comment (guzmandrade-dev) [reply blocks/call-to-action/index.js:1]: 👍

File: inc/wpcli/block-starter/block.json (hunk 1 above)
Comment (guzmandrade-dev) [reply inc/wpcli/block-starter/block.json]: 👍

File: inc/wpcli/block-starter/block.json (hunk 2 above)
Comment (guzmandrade-dev) [reply inc/wpcli/block-starter/block.json]: 👍

File: inc/wpcli/block-starter/editor.js (hunk 3 above)
Comment (guzmandrade-wds) [reply inc/wpcli/block-starter/editor.js:1]: 👍

File: inc/wpcli/block-starter/script.js (hunk 4 above)
Comment (guzmandrade-wds) [reply inc/wpcli/block-starter/script.js:7]: 👍

File: inc/wpcli/class-blocks-scaffold.php (hunk 6 above)
Comment (guzmandrade-dev) [reply inc/wpcli/class-blocks-scaffold.php]: 👍

File: inc/wpcli/class-blocks-scaffold.php (hunk 5 above)
Comment (guzmandrade-dev) [reply inc/wpcli/class-blocks-scaffold.php]: 👍

File: blocks/call-to-action/style.css
Comment (guzmandrade-dev) [reply blocks/call-to-action/style.css:1]: 👍

**Review by khleomix — APPROVED** [review]
LGTM @guzmandrade-wds

**Pull request merged**